src/main/java/com/onarandombox/MultiversePortals/PortalPlayerSession.java

```diff
@@ -282,31 +282,55 @@ public void setTeleportTime(Date date) {
         this.lastTeleportTime = date;
     }
 
-    public boolean allowTeleportViaCooldown(Date date) {
-        return (date.after(new Date(this.lastTeleportTime.getTime() + this.plugin.getCooldownTime())));
+    /**
+     * Get the remaining teleport cooldown time in milliseconds.
+     * If the cooldown is finished, 0 will be returned.
+     *
+     * @return The remaining teleport cooldown time in milliseconds, or 0 if the cooldown is over.
+     */
+    private long getRemainingTeleportCooldown(Date time) {
+        long cooldownEndMs = this.lastTeleportTime.getTime() + this.plugin.getCooldownTime();
+        long timeMs = time.getTime();
+        if (cooldownEndMs > timeMs) {
+            return cooldownEndMs - timeMs;
+        } else {
+            return 0;
+        }
     }
 
-    public String getFriendlyRemainingTimeMessage() {
-        String remaining = "There is a portal " + ChatColor.AQUA + "cooldown" + ChatColor.WHITE + " in effect. Please try again in " + ChatColor.GOLD
-        + Integer.toString((int) this.getRemainingCooldown() / 1000) + "s.";
-        int time = (int) this.getRemainingCooldown() / 1000;
-        // Account for the fact that 0 seconds means 1
-        time++;
-        if (time <= 0) {
-            remaining += "< 1s" + ChatColor.WHITE + ".";
+    /**
+     * Constructs a message informing a player about the teleport cooldown.
+     *
+     * @param cooldownMs The cooldown time in milliseconds.
+     * @return           A message to be sent to a player, informing them about the remaining cooldown time.
+     */
+    private String getCooldownMessage(long cooldownMs) {
+        String remaining = "There is a portal " + ChatColor.AQUA + "cooldown " +
+            ChatColor.WHITE + "in effect. Please try again in " +
+            ChatColor.GOLD;
+
+        if (cooldownMs < 1000) {
+            remaining += "1s";
         } else {
-            remaining += time + "s" + ChatColor.WHITE + ".";
+            remaining += (cooldownMs / 1000) + "s";
         }
```

Comment:
this if..else can be extracted into its own method `formatCooldownTime(int cooldownMs)` then you can just do:

```java
private String getCooldownMessage(long cooldownMs) {
    return "There is a portal " + ChatColor.AQUA + "cooldown " 
            + ChatColor.WHITE + "in effect. Please try again in " 
            + ChatColor.GOLD + formatCooldownTime(cooldownMs)
            + ChatColor.WHITE + ".";
}
```

Reply:
Sure.